server/controler/login.go

```diff
@@ -0,0 +1,81 @@
+package controler
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+type loginReq struct {
+	Username string
+	Password string
+}
+
+type loginResponse struct {
+	Token string `json:"token"`
+}
+
+var minLength = 3
+
+func (env *Env) Login(responseWriter http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		// http.Error(responseWriter,
+		// http.StatusText(http.StatusMethodNotAllowed),
+		// http.StatusMethodNotAllowed)
+		responseWriter.WriteHeader(http.StatusMethodNotAllowed)
```

Comment:
So the difference between the commented section and the current implementation is that `WriteHeader` fills only error code in the response header whilst the previous code was also set some common error message in the response body. A slightly different, I guess we can stay with the header-only version.

Reply:
Will error message be visible in console during "curl" testing?

Reply:
Not sure, worth checking that. The general rule is to return HTTP error code + add a more detailed description in the body of the response if possible.